cornell-dti/course-plan
# Add linguistics major and minor

### Summary 

This pull request adds the linguistics major and minor to CoursePlan

Major
![image](https://user-images.githubusercontent.com/25535093/114282843-59b52800-9a14-11eb-8816-1083bbbe8585.png)

Minor
![image](https://user-images.githubusercontent.com/25535093/114283967-c03d4480-9a1a-11eb-8e05-a858bfa8d0e1.png)

### Test Plan 

1. Confirm the checkers match the information here: https://linguistics.cornell.edu/undergraduate
2. More specifically, add the core classes and confirm they cannot fulfill other linguistics requirements
3. Confirm that additional courses logic holds up (split similarly to liberal studies, has a double countable credit req)
4. Confirm any course can be added to the ancillary skills courses (outside the core courses).
5. Confirm these requirements work for the minor as well.
5. Check that all the text makes sense and is typo-free, and that the urls link where they should

### Notes 

Ran into a few issues when creating this major and minor, and was wondering if we currently have any way of handling these.
- Is there any way we can restrict courses to only display ones that are 3 (or more) credits? Right now I have an 18 credit checker for the additional courses, which is not technically correct, as 2 credit courses could be used to fulfill this requirement when they can't in reality.
- Do we have any way to check a course does not have the "CU-UGR" designation? It shows up in student center under "Course Attribute": https://classes.cornell.edu/browse/roster/FA20/class/LING/4491
- Is there any way to check the restriction that "language courses" cannot be used for additional courses? I don't think so because LING courses can be crosslisted with language course codes and not count as a "language course," but still unsure.
- Right now, one course can be used to fulfill the 3000, 2000, and 1000 course for the minor  but not the major, I think because of the way double counting works (even though none of the requirements are set to allow double counting). Should I rework this requirement for the minor and make it different than the major?
